django_sharding_library/decorators.py

```diff
@@ -38,5 +39,24 @@ def configure(cls):
             setattr(cls, 'django_sharding__shard_group', shard_group)
             setattr(cls, 'django_sharding__is_sharded', True)
 
+            # If the sharded by field is set, we will make our custom manager the default manager.
+            # todo: update docs to indicate that ShardManager should be used as the base class for any custom managers->
+            # for a model that defines a sharded by field
+            # todo: Update docs to include the sharded by field and the get_shard_from_id function
+            if sharded_by_field:
+                from django.db.models import Manager
+                if not isinstance(cls.objects, ShardManager):
+                    if type(cls.objects) == Manager:
```

Comment:
Yes we all prefer that, but in this case it doesnt work: I only want to override the objects attribute if it is the default Django manager. If someone, for example, wrote a custom manager and inherited from the Django default (which everyone does), the isinstance check would return True, and I would overwrite their custom manager with the shard manager. Since I don't want to accidentally remove functionality they are expecting from a custom manager, I want to know and raise an exception telling them to inherit from the ShardManager.

This has come up in my companies project already, i changed it from isinstance to type on purpose.